src/components/custom-field-input-number/custom-field-input-number.jsx

```diff
@@ -0,0 +1,67 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+
+import CustomFieldInputText from '../custom-field-input-text/custom-field-input-text.jsx';
+import CustomFieldInputTextStyles from '../custom-field-input-text/custom-field-input-text.css';
+import useNumberValidator from '../../hooks/number-validator/number-validator.jsx';
+
+function getRootClassName(className, error, disabled) {
+  if (disabled) {
+    return `${className} ${CustomFieldInputTextStyles.disabled}`;
+  }
+
+  if (error) {
+    return `${className} ${CustomFieldInputTextStyles.error}`;
+  }
+
+  return className;
+}
+
+export default function CustomFieldInputNumber(props) {
+  const [input, setInput] = useState('');
+  const valid = props.useValidator(input);
+
+  return (
+    <CustomFieldInputText
+      className={getRootClassName(props.className, props.error, props.disabled)}
+      disabled={props.disabled}
+      error={!valid}
+      helpText={props.helpText}
```

Comment:
I wonder if there is a default error message for non-numeric values. But if we just hook up the backend responses then we probably don't need to worry about it.

Reply:
I worry about having a default error message because we currently have no `i18n` in MDS. I wonder if that's a discussion we should have? Maybe it does have a place here.

Reply:
Good point. I would prefer to just use backend error messaging for now.